src/components/ListManager/ListManager.vue

```diff
@@ -0,0 +1,212 @@
+            }" 
+            :select="() => { selectionMode == 0 ? manager._select(rowData) : {} }"
+            name="listItem"
+          ></slot>
+        </div>
+      </div>
+    </div>
+    <div :class="manager.options?.style?.footer" >
+      <slot name="footer"></slot>
+    </div>
+  </div>
+</template>
+
+<script>
+import { ref, computed } from 'vue'
+
+export default {
+  name: 'ListManager',
+  props: ['options'],
+  setup(props, context) {
+    const options = {
+      handle_row_click: () => {},
+      ...props.options
+    }
+    const resource = ref(null)
+    const newItems = ref([])
+    const selectedItems = ref({})
+    const selectionMode = ref(0)
+
+    const allItemsSelected = computed(() => {
+      if (manager.value.loading) {
+        return false
+      } else {
+        return Object.keys(selectedItems.value).length == manager.value.list.length
+      }
+    })
+
+    const manager = ref({
+      loading: true,
+      resource,
+      options,
+      selectedItems,
+      allItemsSelected,
+      list: [],
+      loadMore: () => {
+        manager.value.loading = true
+        resource.value.update({
+          ...options,
+          start: manager.value.list.length,
+          limit: options.limit
+        })
+      },
+      update: (newOptions) => {
+        clearList()
+        if (newOptions.filters) options.filters = newOptions.filters
+        if (newOptions.order_by) options.order_by = newOptions.order_by
+        resource.value.update({
+          ...options,
+          start: 0,
+          limit: options.limit
+        })
+      },
+      select: (rowData) => {
+        if (selectionMode.value == 1) {
+          selectionMode.value = 2
+        } else if (selectionMode.value == 2) {
+          manager.value._select(rowData)
+        } else {
+          manager.value.options.handle_row_click(rowData)
+        }
+      },
+      selectAll: () => {
+        if (allItemsSelected.value) {
+          selectedItems.value = {}
+        } else {
+          for (let i = 0; i < manager.value.list.length; i++) {
+            selectedItems.value[manager.value.list[i].name] = manager.value.list[i]
+          }
+        }
+      },
+      _select: (rowData) => {
+        if (selectionMode.value == 0) {
+          selectionMode.value = 1
+        }
+        if(rowData.name in selectedItems.value) {
+          delete selectedItems.value[rowData.name]
+          if (Object.keys(selectedItems.value).length == 0) {
+            selectionMode.value = 0
+          }
+        } else {
+          selectedItems.value[rowData.name] = rowData
+        }
+      },
+    })
+    const clearList = () => {
+      manager.value.list = []
+      newItems.value = []
+      selectedItems.value = {}
+    }
+    context.expose({ manager })
+    return {
+      manager,
+      newItems,
+      selectedItems,
+      selectionMode
+    }
+  },
+  mounted() {
+    this.manager.resource = this.$resources.list
+    this.handleRealtimeUpdate()
```

Comment:
Cleanup socket listener on unmount?

Reply:
Also, if you are using `setup` function, you can do the setup/cleanup of socket listener in there.